src/set_theory/surreal.lean

```diff
@@ -173,15 +173,134 @@ theorem mul_zero_equiv (x : pgame) : (x * 0).equiv 0 :=
 /-- `0 * x` has exactly the same moves as `0`. -/
 def zero_mul_relabelling : Π (x : pgame), relabelling (0 * x) 0
 | (mk xl xr xL xR) :=
-⟨by fsplit; rintro (⟨⟨⟩,_⟩ | ⟨⟨⟩,_⟩),
- by fsplit; rintro (⟨⟨⟩,_⟩ | ⟨⟨⟩,_⟩),
- by rintro (⟨⟨⟩,_⟩ | ⟨⟨⟩,_⟩),
+⟨by fsplit; rintro (⟨⟨⟩,_⟩|⟨⟨⟩,_⟩),
+ by fsplit; rintro (⟨⟨⟩,_⟩|⟨⟨⟩,_⟩),
+ by rintro (⟨⟨⟩,_⟩|⟨⟨⟩,_⟩),
  by rintro ⟨⟩⟩
 
 /-- `0 * x` is equivalent to `0`. -/
 theorem zero_mul_equiv (x : pgame) : (0 * x).equiv 0 :=
 (zero_mul_relabelling x).equiv
 
+/-- Local tactic useful for resolving goals involving equivalences between types,
+such as, a × (b ⊕ c) ≃ a × b ⊕ a × c. -/
+meta def try_inl_inr : tactic unit :=
+`[assumption]
+    <|> (do `[apply sum.inl], try_inl_inr )
+    <|> (do `[apply sum.inr], try_inl_inr )
+    <|> (do `[apply prod.mk], try_inl_inr, try_inl_inr )
+
+local infix ` ≈ ` := pgame.equiv
+
+lemma left_distrib_equiv_aux {a b c d e : pgame} : (a + b) + (c + d) - (e + b) ≈ a + c - e + d :=
```

Comment:
I don't understand why this isn't trivial.

We already know `game` is an honest `add_comm_group`. So why don't you just turn this into an equality goal in `game`, and run `abel`?

Reply:
I tried this method. When I do `import set_theory.game tactics.abel` in `surreal.lean` the `calc` proofs in the file breakdown. I get the following error message: 

```lean
type mismatch at application
  has_add.add w
term
  w
has type
  pgame : Type (u+1)
but is expected to have type
  game : Type (?+1)
state:
w x y z : pgame,
ow : w.numeric,
ox : x.numeric,
oy : y.numeric,
oz : z.numeric,
ix : x.left_moves,
hix : w ≤ x.move_left ix,
iz : z.left_moves,
hiz : y ≤ z.move_left iz
⊢ w + y ≤ x.move_left ix + z
```
I posted it on [Zulip](https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/Surreal.20numbers/near/236939373) but couldn't figure out what's going on, so I reverted back to a proof without using `set_theory.game`.

This is the commit (6b6366b) before the revert: https://github.com/leanprover-community/mathlib/pull/7440/commits/6b6366b351f3e3c242e620aa0c39a4dc1eca8c10

Reply:
Also, something goes wrong with the `refl`s but replacing them with `apply le_refl` works.
I couldn't find a similar hack for the `calc` proofs.

Reply:
Removing `@[refl]` from `game.le_refl` and `@[trans]` from `game.le_trans` resolves this issue. I can use this hack for now but I don't quite understand what's going on and am worried that something else might break in the future.

Reply:
I implemented this and the proof looks much nicer now. I hope this doesn't cause any problems in the future.